src/main/java/com/bmuschko/gradle/docker/internal/ConventionPluginHelper.java

```diff
@@ -15,8 +18,18 @@ private ConventionPluginHelper() {
     }
 
     public static SourceSetOutput getMainJavaSourceSetOutput(Project project) {
-        JavaPluginExtension javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
-        return javaPluginExtension.getSourceSets().getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput();
+        SourceSetContainer sourceSets = getMainJavaSourceSetContainer(project);
+        return sourceSets.getByName(SourceSet.MAIN_SOURCE_SET_NAME).getOutput();
+    }
+
+    private static SourceSetContainer getMainJavaSourceSetContainer(Project project) {
+        try {
+            JavaPluginExtension javaPluginExtension = project.getExtensions().getByType(JavaPluginExtension.class);
```

Comment:
This class was introduced in Gradle 4.10: https://docs.gradle.org/current/javadoc/org/gradle/api/plugins/JavaPluginExtension.html. There's no need to catch this case where the class doesn't exist. We do not support pre-4.10 Gradle versions anymore.